src/QmlControls/QGCButton.qml

```diff
@@ -11,32 +12,50 @@ Button {
 
     property var __qgcPal: QGCPalette { colorGroupEnabled: enabled }
 
-    property bool __showHighlight: pressed | hovered | checked
+    property bool __showHighlight: (pressed | hovered | checked) && !__forceHoverOff
 
-    style: ButtonStyle {
-/*
-            background: Rectangle {
-                implicitWidth: 100
-                implicitHeight: 25
-                color: __showHighlight ?
-                    control.__qgcPal.buttonHighlight :
-                    (primary ? control.__qgcPal.primaryButton : control.__qgcPal.button)
-            }
+    // This fixes the issue with button hover where if a Button is near the edge oa QQuickWidget you can
+    // move the mouse fast enough such that the MouseArea does not trigger an onExited. This is turn
+    // cause the hover property to not be cleared correctly.
 
-            label: Text {
-                width: parent.width
-                height: parent.height
+    property bool __forceHoverOff: false
 
-                verticalAlignment: TextEdit.AlignVCenter
-                horizontalAlignment: TextEdit.AlignHCenter
+    property int __lastMouseX:  __behavior.mouseX
+    property int __lastMouseY:  __behavior.mouseY
+    property int __lastCursorX: 0
+    property int __lastCursorY: 0
 
-                text: control.text
-                color: __showHighlight ?
-                    control.__qgcPal.buttonHighlightText :
-                    (primary ? control.__qgcPal.primaryButtonText : control.__qgcPal.buttonText)
+    property MousePosition __mousePosition: MousePosition { }
+
+    Connections {
+        target: __behavior
+        onMouseXChanged: {
+            __lastCursorX = __mousePosition.mouseX
+            __lastCursorY = __mousePosition.mouseY
+        }
+        onMouseYChanged: {
+            __lastCursorX = __mousePosition.mouseX
+            __lastCursorY = __mousePosition.mouseY
+        }
+        onEntered: { __forceHoverOff; false; hoverTimer.start() }
+        onExited: { __forceHoverOff; false; hoverTimer.stop() }
+    }
+
+    Timer {
+        id:         hoverTimer
+        interval:   250
+        repeat:     true
+
+        onTriggered: {
+            if (__lastCursorX != __mousePosition.mouseX || __lastCursorY != __mousePosition.mouseY) {
+                __forceHoverOff = true
+            } else {
+                __forceHoverOff = false
             }
```

Comment:
I haven't tried it, but I don't see where you map both mouse coordinates to world. Are you comparing the MouseArea coordinates (where its 0,0 is the 0,0 within the control) and the cursor position returned by the main window (which has its 0,0 based on the main window 0,0)?